lightningd/options.c

```diff
@@ -759,6 +759,23 @@ static void opt_parse_from_config(struct lightningd *ld, bool early)
 		else {
 			/* Only valid forms are "foo" and "foo=bar" */
 			all_args[i] = tal_fmt(all_args, "--%s", lines[i]);
+
+			char *pos = strchr(all_args[i], '#');
```

Comment:
Error in follow config file:
```
   bitcoin-rpcpassword=xxx#pass  # xxx
```
Maybe the fix should add to ccan/ccan/opt/parse.c

Reply:
ccan fixes should be integrated into ccan directly https://github.com/rustyrussell/ccan/

Reply:
tnx for suggestions, i'll find a cleaner approach.

Core doesnt allow such passwords, so neither should we:

> bitcoin-rpcpassword=xxx#pass  # xxx

(16115)gorazd@gorazd-MS-7C37:~/Projects/bitcoin$ ./src/bitcoind
Error: Error reading configuration file: parse error on line 5, using # in rpcpassword can be ambiguous and should be avoided

Reply:
tnx ill find a cleaner approach

i see now how Core doesnt allow neither # in passwords nor trailing comments after passwords:

> Error: Error reading configuration file: parse error on line 6, using # in rpcpassword can be ambiguous and should be avoided